zoom.js

```diff
@@ -0,0 +1,236 @@
+    $('#wc-container-left').css('max-height', "clac(300% - var(--comment-area-height) - 30px)");
+    $(".main").append(
+      '<highlight-chat></highlight-chat><button class="btn-clear">CLEAR</button>' +
+      '<span style="font-size: 0.7em">Aspect Ratio: <span id="aspect-ratio"></span></span>'
+    );
+    popOutChat();
+  }, 2000)
+}
+
+// This is an individual message.
+const ZOOM_CHAT_MESSAGE_SELECTOR = '.chat-message-text-content';
+// Each user's message is within a container, which has their name and avatar.
+// Multiple messages can be in one container.
+const ZOOM_CHAT_USER_CONTAINER = 'chat-item__chat-info';
+
+// TODO: This is a hack.
+window.location.href.indexOf('zoom') > 0 && setupWindow();
+window.location.href.indexOf('zoom') > 0 && $("body")
+  .unbind("click")
+  .on(
+    "click",
+    ZOOM_CHAT_MESSAGE_SELECTOR,
+    function () {
+      let $this = $(this);
+      let container = $this.parentsUntil(ZOOM_CHAT_USER_CONTAINER);
+
+      $(".hl-c-cont").remove();
+      var chatname = container.find(".chat-item__sender").text();
+
+      var chatmessage = $this.html();
+      // var chatmembership = $this
+      //   .find(".yt-live-chat-membership-item-renderer #header-subtext")
+      //   .html();
+      var useravatar = null;
+      let avatar = container.find(".chat-item__chat-info-msg-avatar");
+      if (avatar.length > 0) {
+        useravatar = avatar.attr('src');
+      }
+
+      $this.addClass("show-comment");
+
+      // TODO: Extract this.
+      var backgroundColor = "";
+      var textColor = "";
+      if (
+        this.style.getPropertyValue("--yt-live-chat-paid-message-primary-color")
+      ) {
+        backgroundColor =
+          "background-color: " +
+          this.style.getPropertyValue(
+            "--yt-live-chat-paid-message-primary-color"
+          ) +
+          ";";
+        textColor = "color: #111;";
+      }
+
+      // TODO: Sponsors don't apply in Zoom?
+      if (this.style.getPropertyValue("--yt-live-chat-sponsor-color")) {
+        backgroundColor =
+          "background-color: " +
+          this.style.getPropertyValue("--yt-live-chat-sponsor-color") +
+          ";";
+        textColor = "color: #111;";
+      }
+
+      $("highlight-chat")
+        .append(renderChatMessage(chatmessage, chatname, useravatar, {
+          hasDonation: '',
+          hasMembership: '',
+          chatbadges: '',
+          style: `${backgroundColor} ${textColor}`
+        }))
+        .delay(10)
+        .queue(function (next) {
+          $(".hl-c-cont").removeClass("fadeout");
+          next();
+        });
+    }
+  );
+
+// TODO: These two chunks can be extracted.
+$("body").on("click", ".btn-clear", function () {
+  $(".hl-c-cont")
+    .addClass("fadeout")
+    .delay(300)
+    .queue(function () {
+      $(".hl-c-cont").remove().dequeue();
+    });
+});
+
+
+function renderChatMessage(message, name, image, opts ) {
```

Comment:
This fell behind any recent changes made, but I this was an attempt to define a function that could live elsewhere.